diff -Naur spacefm-2.1.0.orig/meson.build spacefm-2.1.0/meson.build
--- spacefm-2.1.0.orig/meson.build	2022-08-17 19:04:16.788078104 +0300
+++ spacefm-2.1.0/meson.build	2022-08-17 19:05:44.209022628 +0300
@@ -94,6 +94,7 @@
 
 if get_option('sn')
   pre_args += '-DHAVE_SN'
+  dep_sn = dependency('libstartup-notification-1.0', method : 'pkg-config')
 endif
 
 if get_option('deprecated-hw')
@@ -203,6 +204,7 @@
     dep_glib,
     dep_gtk,
     dep_x11,
+    dep_sn,
     lib_math,
     lib_libbsd,
     lib_udev,
